src/common/string/filter/sequence-matching.ts

```diff
@@ -0,0 +1,65 @@
+import { fuzzyScore, isPatternInWord } from "./filter";
+
+/**
+ * Determine whether a sequence of letters exists in another string,
+ *   in that order, allowing for skipping. Ex: "chdr" exists in "chandelier")
+ *
+ * filter => sequence of letters
+ * word => Word to check for sequence
+ *
+ * returns either:
+ *    number => if word contains sequence, return a score
+ *    string => if word is empty, return the word (to allow for alphabetical sorting of all available words)
```

Comment:
Not sure we should do this, is `fuzzySequentialMatch` responsible for this?

Reply:
I put this behavior in so that an empty filter string (like when you first launch dialog) wouldn't return `undefined` from the scorer, and give us a clue that we now need to sort alphabetically rather than by score.

But now that I think about it, I guess we could take care of all that behavior inside `ha-quick-bar`. We could have quick bar detect an empty filter string and not bother sending to filter.

Reply:
Removed